src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasRelationSlicer.java

```diff
@@ -328,8 +367,62 @@ private long createNewPointIdentifier(
         else
         {
             final long identifier = pointIdentifierFactory.nextIdentifier();
-            return this.partiallySlicedRawAtlas.point(identifier) == null ? identifier
-                    : createNewPointIdentifier(pointIdentifierFactory, coordinate);
+            return this.partiallySlicedRawAtlas.point(identifier) == null
+                    && !this.slicedRelationChanges.getCreatedPoints().containsKey(identifier)
+                            ? identifier
+                            : createNewPointIdentifier(pointIdentifierFactory, coordinate);
+        }
+    }
+
+    /**
+     * Cuts out data outside the shard boundary, since we don't need any data beyond that.
+     *
+     * @param atlas
+     *            The {@link Atlas} file we need to trim
+     * @return the {@link Atlas} for the bounds of the input shard
+     */
+    private Atlas cutSubAtlasForOriginalShard(final Atlas atlas)
+    {
+        try
+        {
+            if (this.initialShard != null)
+            {
+
+                final Atlas countrySubAtlas = atlas.subAtlas(entity ->
```

Comment:
Maybe put that lambda in a variable a line above for readability? I was initially surprised by the SOFT_CUT in the middle line 408 :)

Reply:
good idea. I also improved the comments here explaining why there are two cuts, and the purpose of SILK_CUT vs SOFT_CUT